src/smart-components/user/user.js

```diff
@@ -0,0 +1,229 @@
+        // eslint-disable-next-line new-cap
+        debouncedFetch = AwesomeDebouncePromise(fetchRoles, 500);
+
+    }, []);
+
+    useEffect(() => {
+        if (!isLoading) {
+            setRows(createRows(roles.data));
+        }
+    }, [ roles ]);
+
+    useEffect(() => {
+        setRows(createRows(roles.data));
+    }, [ rolesWithAccess ]);
+
+    useEffect(() => {
+        debouncedFetch({ limit: perPage, offset: perPage * (page - 1), name: filter, addFields: [ 'groups_in' ], username });
+    }, [ filter ]);
+
+    useEffect(() => {
+        fetchRoles({ limit: perPage, offset: perPage * (page - 1), name: filter, addFields: [ 'groups_in' ], username });
+    }, [ page, perPage ]);
+
+    const onExpand = (event, rowIndex, colIndex, isOpen, rowData) => {
+        const r = [ ...rows ];
+        if (!isOpen) {
+            setExpanded({ ...expanded, [rowData.uuid]: colIndex });
+            // Permissions
+            if (colIndex === 2) {
+                fetchRoleForUser(rowData.uuid);
+            }
+
+            // set all other expanded cells false in this row if we are expanding
+            r[rowIndex].cells.forEach(cell => {
+                if (cell.props) {cell.props.isOpen = false;}
+            });
+            r[rowIndex].cells[colIndex].props.isOpen = true;
+            r[rowIndex].isOpen = true;
+        } else {
+            setExpanded({ ...expanded, [rowData.uuid]: -1 });
+            r[rowIndex].cells[colIndex].props.isOpen = false;
+            r[rowIndex].isOpen = r[rowIndex].cells.some(cell => cell.props && cell.props.isOpen);
+        }
+
+        setRows(r);
+    };
+
+    const onSetPage = (_e, page) => setPage(page);
+    const onPerPageSelect = (_e, perPage) => setPerPage(perPage);
+
+    return (<Stack >
+            <StackItem>
+                <TopToolbar paddingBottm={ false }>
+                    <TopToolbarTitle
+                    title={ username }
+                    description={ `${username}'s roles, groups and permissions.` }
+                    />
+                </TopToolbar>
+            </StackItem>
+            <StackItem>
+                <Section type="content" id={ 'user-detail2' }>
+                <PrimaryToolbar
+                    filterConfig={ { items: [{ label: 'Role',
+                        filterValues: {
+                            value: filter,
+                            placeholder: 'Filter by role name',
+                            onChange: (e, selected) =>setFilter(selected)
+                        }
+                    }]} }
+                    pagination={ { itemCount: roles.meta.count, page, perPage, onSetPage, onPerPageSelect } }
+                />
+                { isLoading
+                ? <ListLoader />
+                : <Table
+                    className='ins-c-activity-table'
+                    aria-label="Collapsible table"
+                    onExpand={ onExpand }
```

Comment:
I'd extend the `TableToolbarView` to consume `onExpand` function this way we can easilly use this table variant without the need to copy this code all over the places.